tasks/libs/github_actions_tools.py

```diff
@@ -86,20 +90,35 @@ def trigger_macos_workflow(
     # Thus the following hack: Send an id as input when creating a workflow on Github. The worklow will use the id and put it in the name of one of its jobs.
     # We then fetch workflows and check if it contains the id in its job name.
 
+    # Adding another hack to check if the workflow is not waiting a concurrency to be solved before generating the workflow jobs
+    might_be_waiting = set()
+
+    MAX_WAITING_CONCURRENCY_RETRIES = 3  # Retries for up to 1h30
```

Comment:
As the fetching can take up to 5mins
```suggestion
    MAX_WAITING_CONCURRENCY_RETRIES = 3  # Retries for up to 1h45
```